### PR TITLE
Replace CLang file hashing with .st_mtime

### PR DESCRIPTION
**Pull request by ClaudiuHKS**
Replace CLang file hashing with .st_mtime for performance.
Also, fix a problem in CLangMngr::MergeDefinitionFile.

Thanks Arkshine for their tutorials of using StringHashMap.
He really contributed very hard on this CLang change.

### CODE REVIEW

**Comment by Arkshine** [comment]
Much better. (I'm one who asked him to redo PR/branch, because previous branch was completely fucked up, and I failed to help him to clean it up. It was just more fast and simple to redo the PR).

File: amxmodx/CLang.h [hunk 1]
```diff
@@ -118,7 +113,7 @@ class CLangMngr
 	
 	LangVec m_Languages;
 
-	CVector<md5Pair *> FileList;
+	StringHashMap<time_t> FileList;
 	CVector<String *> KeyList;
 	THash<String, keytbl_val> KeyTable;
```
Comment (xPaw) [inline amxmodx/CLang.h]: You should try merging KeyList/KeyTable into a single HashTable from AMTL, which will mean tinyhash class will no longer be needed.

Comment (Arkshine) [inline amxmodx/CLang.h]: I would do that in another PR. The following patch is just to fix the shit code on this specific chunk of code.
I will make a PR to clean the whole class.

Comment (ClaudiuHKS) [inline amxmodx/CLang.h]: Exactly. xPaw, I'm not familiarized with amtl. Really, I'd appreciate Arkshine's hands doing these amtl implementations. I will help any other ways, if needed, for now.

Agree, that CLang code really needed a change.

**Comment by Arkshine** [comment]
`stat` should be fine under windows, since both `stat` and `_stat` seems to point to the the same internal function.
 
:palm_tree:

**Comment by Nextra** [comment]
ltgm then

**Pull request merged**